src/apps/deskflow-gui/MainWindow.cpp

```diff
@@ -466,7 +484,27 @@ void MainWindow::updateSize()
 
 void MainWindow::showMyFingerprint()
 {
-  QMessageBox::information(this, "TLS fingerprint", TlsFingerprint::local().readFirst());
+  auto localPath = QStringLiteral("%1/%2").arg(getTlsPath(), kFingerprintLocalFilename).toStdString();
+  if (!QFile::exists(QString::fromStdString(localPath))) {
+    if (regenerateLocalFingerprints())
+      showMyFingerprint();
+    return;
+  }
+
+  deskflow::FingerprintDatabase db;
```

Comment:
I realised that we use `deskflow::` in 16 places in this file. Perhaps it'd be better to have `using namespace deskflow;` at the top.

```suggestion
  FingerprintDatabase db;
```

Reply:
No doing that kinds defeats the point of a namespace..